### DISCUSSION TITLE
Error 9, and 22 and over

### DISCUSSION THREAD

**Discussion by Avolrrabi** [fra589/grbl-Mega-5X#275] [Q&A] [answered]
Thank you for your code

 I apologize in advance for bad English, I hope it will be possible to understand

 The problem itself:
 I am assembling a CNC machine and chose your firmware as the most suitable for my tasks.  All this will work with grbl gru
 Electronic part:
 - Firmware is on Arduino mega 2560
 - Together with RAMPS 1.6+
 - 3 nema 17 stepper motors are connected (nothing more is known about them, only that they used to be in Creality 3D)
 - 3 DRV8825 drivers
 - Power supply 12V 10.5A
 - There is nothing else (horse switches of sensors, etc. Nothing at all)
 If you run sketches on the Arduino without firmware (specifically, it was a sketch from the ramps 1.4 examples, meaning just the rotation of the motors in one direction and the other when power is applied), then everything works.  Engines spin, drivers do not squeak, etc.  From this I conclude that everything is assembled correctly, Arduino is working, ramps are working, drivers and motors are working.

 Having problems installing grbl mega 5x.
 Arduino responds to commands $$, $#, $G and others.  At the same time, it does not react to an attempt to start the motors (both through the Arduino IDE with G01 commands, and through GrblGru by any means)
 Gives error 9, error 22

 Results of commands $$, $#, $G

$0=10
$1=128
$2=0
$3=0
$4=0
$5=0
$6=0
$10=0
$11=0.020
$12=0.002
$13=0
$20=0
$21=0
$22=0
$23=0
$24=500.000
$25=2000.000
$26=250
$27=5.000
$30=1000
$31=0
$32=0
$100=200.000
$101=400.000
$102=400.000
$103=8.889
$104=8.889
$110=20000.000
$111=750.000
$112=750.000
$113=1440.000
$114=1440.000
$120=5.000
$121=50.000
$122=50.000
$123=50.000
$124=50.000
$130=2222.000
$131=200.000
$132=200.000
$133=360.000
$134=180.000
ok
[G54:0.000,0.000,0.000,0.000,0.000]
[G55:0.000,0.000,0.000,0.000,0.000]
[G56:0.000,0.000,0.000,0.000,0.000]
[G57:0.000,0.000,0.000,0.000,0.000]
[G58:0.000,0.000,0.000,0.000,0.000]
[G59:0.000,0.000,0.000,0.000,0.000]
[G28:0.000,0.000,0.000,0.000,0.000]
[G30:0.000,0.000,0.000,0.000,0.000]
[G92:0.000,0.000,0.000,0.000,0.000]
[TLO:0.000]
[PRB:0.000,0.000,0.000,0.000,0.000:0]
ok
[GC:G0 G54 G17 G21 G90 G94 M5 M9 T0 F0 S0]
ok

 Please help me 

![image](https://user-images.githubusercontent.com/108528695/176894955-ad218094-1ea0-4aa4-8219-6e1726a0885e.jpeg)

**Comment by fra589** [comment, marked as answer]
Hi @Avolrrabi,  
  
Error 9 is "G-code lock" it means that the G-code commands are locked out during alarm or jog state.
You can verify the Grbl status with the `?` command.

Since you don't have limit switches, you can't do the initial homing. So, you will have to disable all limits and homing in parameters:
```
$20=0
$21=0
$22=0
```
This is what you do, good! :smiley: 
Without this, the system will be in Alarm state at power on and you will have the error 9 unless you kill the Alarm state with the `$X` command.

The GCode G1 move is a linear working move at working speed. The error 22 is given when you forget to give the speed to this move.
The correct syntax for a G1 move is:
(example)
`G1F150X10Y20Z5`
Depending of the motion mode (G90/G91) and the units mode (G20/G21)...
If the motion mode is in absolute distance mode (G90) and unit in mm (G21), this will give a move at 150 mm/sec to the position X=10mm, Y=20mm and Z=5mm.

For more details about GCode syntaxes and CNC philosophy, I recommend you to go to the best documentation for it:
http://linuxcnc.org/docs/html/index.html

Reading the Wiki https://github.com/fra589/grbl-Mega-5X/wiki is also always a good ID. :smiley: 

@++;
Gauthier.

**Reply by Avolrrabi** [reply]
Thanks for the quick response.  

You are gorgeous!  I send a video of the working machine 

https://user-images.githubusercontent.com/108528695/176950799-60f572a5-42a3-4753-852b-e760f0377bdc.MOV
